index.html

```diff
@@ -43,10 +43,32 @@ <h1 class="card-title">
           <span id="date-display"></span>
         </div>
 
-        <div class="card-body text-center">
+        <div class="card-body">
           <div class="row">
-            <div class="col-1"></div>
             <div class="col-7">
+              <div class="input-group">
+                <input
+                  id="city-input"
+                  class="form-control"
+                  type="text"
+                  placeholder="city"
+                />
+                <div class="input-group-append">
+                  <button
+                    id="search-weather"
+                    class="btn btn-outline-secondary"
+                    type="submit"
+                  >
+                    <i class="fas fa-search-location"></i>
+                  </button>
```

Comment:
Wrap buttons and inputs like these inside a `form`:
```html
<form id="city-search-form">
<input ...>
<button ...>
</form>
```

Then, add an event listener to the `sumbit` event of that form. Inside the listener, you have to prevent the default action of the form (reloading the page), and then you can do the same that you already do in your `click` event listener on the button:

```javascript
document.querySelector('#city-search-form').addEventListener('submit', (event) => {
    event.preventDefault();
    // continue as usual
});
```

The big advantage of that approach is that this way, you automatically listen to clicks on the button, enter key presses in the input element, and native submit buttons in the touch-screen keyboards of mobile browsers

Reply:
Another reason (which is irrelevant in your case right now, but you may implement this as well) why form submit-events are better: If you had validation rules inside your input elements (for example `required`, `maxlength`, `pattern` etc), the `submit` event would automatically validate those and display a message if those validations fail.

If you want, you can try this out yourself: Add the attribute `required` or `minlength="3"` to your input and see what happens

Reply:
Thanks, that was really useful :), I was wondering before how to make it work by making entering a city and pressing enter, now is clear.
With what I'm currently working on,  I realized that I have another issue to work with ... the sunset time and sunrise time from places in different time zone is off. Maybe that could be work in a new branch later.